tiledb/api/c_api/profile/test/unit_capi_profile.cc

```diff
@@ -0,0 +1,179 @@
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    // Ensure nullptr resolves to default name internally.
+    tiledb_string_t* name;
+    rc = tiledb_profile_get_name(profile, &name, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    const char* out_ptr;
+    size_t out_length;
+    rc = tiledb_string_view(name, &out_ptr, &out_length);
+    REQUIRE(rc == TILEDB_OK);
+    std::string out_str(out_ptr, out_length);
+    CHECK(out_str == std::string(fx.name_));
+    REQUIRE_NOTHROW(tiledb_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("empty homedir") {
+    rc = tiledb_profile_alloc(fx.name_, "", &profile, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null homedir") {
+    // Normal expected use-case. Internally resolves to default homedir.
+    rc = tiledb_profile_alloc(fx.name_, nullptr, &profile, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    tiledb_string_t* homedir;
+    rc = tiledb_profile_get_homedir(profile, &homedir, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    // Validate homedir is non-empty. The path may not be resolved when using
+    // the default homedir (per the RestProfile::homedir() invariant).
+    const char* out_ptr;
+    size_t out_length;
+    rc = tiledb_string_view(homedir, &out_ptr, &out_length);
+    REQUIRE(rc == TILEDB_OK);
+    std::string out_str(out_ptr, out_length);
+    CHECK(out_str == tiledb::common::filesystem::home_directory());
+    REQUIRE_NOTHROW(tiledb_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("null profile") {
+    rc = tiledb_profile_alloc(fx.name_, homedir_, nullptr, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+}
+
+TEST_CASE("C API: tiledb_profile_free argument validation", "[capi][profile]") {
+  CAPINProfileFx fx;
+  tiledb_profile_t* profile;
+  tiledb_error_t* err = nullptr;
+  auto rc = tiledb_profile_alloc(
+      fx.name_, fx.tempdir_.path().c_str(), &profile, &err);
+  REQUIRE(tiledb_status(rc) == TILEDB_OK);
+  SECTION("success") {
+    REQUIRE_NOTHROW(tiledb_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("null profile") {
+    tiledb_profile_free(nullptr);
+  }
+}
+
+TEST_CASE(
+    "C API: tiledb_profile_get_name argument validation", "[capi][profile]") {
+  CAPINProfileFx fx;
+  capi_return_t rc;
+  tiledb_error_t* err = nullptr;
+  ordinary_profile x{fx.name_, fx.tempdir_.path().c_str()};
+  tiledb_string_t* name;
+  SECTION("success") {
+    rc = tiledb_profile_get_name(x.profile, &name, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    const char* out_ptr;
+    size_t out_length;
+    rc = tiledb_string_view(name, &out_ptr, &out_length);
+    REQUIRE(rc == TILEDB_OK);
+    std::string out_str(out_ptr, out_length);
+    REQUIRE(out_str == std::string(fx.name_));
+  }
+  SECTION("null profile") {
+    rc = tiledb_profile_get_name(nullptr, &name, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null name") {
```

Comment:
Is the `err` argument allowed to be NULL?  Can you add another section demonstrated the expected behavior?

Reply:
This results in a `TileDB internal: Error argument may not be a null pointer`, which seems logical given the implementation of `api_entry_error`. If we don't want to pass an `err` argument, we should instead use `api_entry_plain`.

Reply:
I think having the error argument is correct, but I would like to see a test case which passes `nullptr`. Everything I see now passes `&err`.

Reply:
As I mentioned in my previous comment, this results in a `TileDB internal: Error argument may not be a null pointer`. If we don't want to pass an `err` argument, we should use `api_entry_plain` instead. `api_entry_error` was used based on a previous comment.